src/lib/pages/backoffice/DocumentRequest/DocumentRequestDetails/DocumentRequestSteps/DocumentRequestStepPanels/ChooseDocumentStepPanel/ChooseDocumentStepPanel.js

```diff
@@ -15,36 +15,33 @@ export default class ChooseDocumentStepPanel extends Component {
   };
 
   goToCreateDocumentAndPrefill = (docReq) => {
-    const data = {};
-    const authors = _get(docReq, 'authors');
-    if (authors) {
-      data['authors'] = [{ full_name: _get(docReq, 'authors') }];
-    }
-    const journalTitle = _get(docReq, 'journal_title', '');
-    const issue = _get(docReq, 'issue', '');
-    const volume = _get(docReq, 'volume', '');
-    if (journalTitle || issue || volume) {
-      data['publication_info'] = [
-        {
-          journal_title: journalTitle,
-          journal_issue: issue,
-          journal_volume: volume,
-        },
-      ];
-    }
-    const isbn = _get(docReq, 'isbn');
-    if (isbn) {
-      data['identifiers'] = [{ scheme: 'ISBN', value: isbn }];
-    }
+    const {
+      authors,
+      journal_title: journalTitle,
+      issue,
+      volume,
+      isbn,
+      edition,
+      publication_year,
+      publisher,
+    } = docReq;
 
-    const edition = _get(docReq, 'edition');
-    if (edition) {
-      data['edition'] = edition;
-    }
-    const publicationYear = _get(docReq, 'publication_year');
-    if (publicationYear) {
-      data['publication_year'] = `${publicationYear}`; // it has to be a string
-    }
+    const data = {
+      authors: authors && [{ full_name: authors }],
```

Comment:
this is not correct according to the behaviour that was there before. The `authors` key now will be always there even if it's `undefined` compared to before that was added only if defined.

Reply:
ah ok! I thought it would not matter much because if there is no value given it would be undefined anyhow

Reply:
That depends on how this value is checked on other parts of the code. Maybe @kprzerwa can confirm if including in the `data` object only the defined values are mandatory or not :)